### DISCUSSION TITLE
ip address from callback

### DISCUSSION THREAD

**Discussion by titov32** [pymodbus-dev/pymodbus#1958] [Q&A] [answered]
I have reviewed an example of using a callback to request data from [example](https://github.com/pymodbus-dev/pymodbus/blob/dev/examples/server_callback.py) and it's not entirely clear whether it's possible to obtain the IP address of the client that requested the data and the IP address of the server providing the data.

**Comment by janiversen** [comment, marked as answer]
Of course you can get the ip address but it is not provided in the call. The ip address of the server if what you configured.

To get the IP address you need to dig down in the transport code and examine the socket.
